PyInstaller/building/makespec.py

```diff
@@ -440,6 +441,15 @@ def __add_options(parser):
             "\n"
         ),
     )
+    g.add_argument(
+        '--optimize',
+        dest='optimize',
+        metavar='LEVEL',
+        type=int,
+        choices={-1, 0, 1, 2},
```

Comment:
I'm not sure about a visible choice of -1. I don't want people inferring that -1 is an optimization level.

Reply:
Do you mean that -1 should not be an allowed value here? (we could remove it, since the implemented verbatim pass-through for -1 has limited usability due to lack of corresponding `O` options for EXE). Because `--help` message does not seem to list the choices, presumably due to `metavar` being specified (which was more accidental than intentional).

Reply:
> Because --help message does not seem to list the choices,

In that case, I guess it doesn't matter. It appears in the error message if you do something like `--optimize=6` but I think we can live with that.